examples/stubbing-spying__route2/cypress/integration/stub-fetch-spec.js

```diff
@@ -0,0 +1,167 @@
+/// <reference types="Cypress" />
+
+describe('route2', () => {
+  context('stubbing', function () {
+    it('shows no Response message', () => {
+      // stub server response with []
+      // for now have to stringify empty arrays
+      // https://github.com/cypress-io/cypress/issues/8532
+      cy.route2('/favorite-fruits', JSON.stringify([]))
+      cy.visit('/')
+      cy.contains('No favorites').should('be.visible')
+    })
+
+    it('stubs fetch to test loading indicator', () => {
+      cy.route2('/favorite-fruits', (req) => {
+        req.reply((res) => {
+          res.delay(2000).send(['Pineapple 🍍'])
+        })
+      })
+
+      cy.visit('/')
+      // at first, the app is showing the loading indicator
+      cy.get('.loader').should('be.visible')
+      // once the promise is resolved, the loading indicator goes away
+      cy.get('.loader').should('not.exist')
+      cy.contains('li', 'Pineapple 🍍')
+    })
+
+    // A big advantage of controlling the response is we can test
+    // how our app handles a slow response, which normally might be
+    // difficult against a fast development server
+    it('shows loader while fetching fruits', function () {
+    // stub the XHR request from the app
+      cy.route2('/favorite-fruits', (req) => {
+        req.reply((res) => {
+          // hmm, every time we want to return an empty list
+          // we need to stringify it, otherwise the stub does not ... stub
+          res.delay(1000).send(JSON.stringify([]))
+        })
+      })
+
+      cy.visit('/')
+      cy.get('.loader').should('be.visible')
+
+      // once the network call finishes, the loader goes away
+      cy.get('.loader').should('not.exist')
+      cy.contains('.favorite-fruits', 'No favorites')
+    })
+
+    // NOTE: this does not work: cannot use cy commands inside the request handler
+    it.skip('shows loading indicator (alternative)', function () {
```

Comment:
FYI the user wanted to do this, but we cannot (promise inside the code, etc) @flotwig

Reply:
@bahmutov see this issue: https://github.com/cypress-io/cypress/issues/7667